src/unix/udp.c

```diff
@@ -409,6 +411,19 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
   QUEUE* q;
   ssize_t size;
 
+  if(handle->flags & UV_HANDLE_UDP_CONNECT_IN_PROGRESS) {
+    int sock_error = uv__udp_check_socket_connected(handle);
+    if(sock_error == 0) {
+        handle->flags &= ~UV_HANDLE_UDP_CONNECT_IN_PROGRESS;
+        handle->flags |= UV_HANDLE_UDP_CONNECTED;
+    } else if(sock_error == UV__ERR(EINPROGRESS)) {
+        uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
+        return;
+    } else {
```

Comment:
No need to start watching for POLLOUT events as it'll be automatically done when the write_queue is not empty